x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractFieldScript.java

```diff
@@ -97,7 +97,7 @@ public final void setDocument(int docId) {
     }
 
     protected final List<Object> extractFromSource(String path) {
-        return XContentMapValues.extractRawValues(path, leafSearchLookup.source().loadSourceIfNeeded());
+        return XContentMapValues.extractRawValues(path, leafSearchLookup.source());
```

Comment:
This isn't critical, but I found the original strategy clearer and more direct as it doesn't rely on the fact `SourceLookup` implements `Map`. This and the instance below would become `leafSearchLookup.source().source()`.

Reply:
I have a separate PR which makes LeafSearchLookup just return Maps, as we are going to want to have lookups which don't rely on the index for calculated fields.  So I'm leaning in the opposite direction here - we want to rely explicitly on the fact that SourceLookup implements Map, as it means that we can in some cases just hand it a pre-built map that doesn't necessarily have to be read from the index.

Reply:
See #67415

Reply:
I find it hard to determine what's best without the big picture (of moving runtime fields to index time). Since it's a tiny change, maybe we could do the straightforward refactor for now, just replacing `loadSourceIfNeeded()` with `source()`, then discuss in future PRs?

Reply:
Apologies for not looking at this soon. I echo @jtibshirani concerns here. The loadSourceIfNeeded -> source change seems like a positive though. Maybe we could avoid anything else until the design for rt at index time is a bit more fleshed out.